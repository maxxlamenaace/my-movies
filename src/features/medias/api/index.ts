import { getMediaCredits } from './get-media-credits/get-media-credits';
import { getMediaDetails } from './get-media-details/get-media-details';
import { getMediaGenres } from './get-media-genres/get-media-genres';
import { getMediaImages } from './get-media-images/get-media-images';
import { getMediaRecommendations } from './get-media-recommendations/get-media-recommendations';
import { getMediaVideos } from './get-media-videos/get-media-videos';
import { getMediasList } from './get-medias-list/get-medias-list';

export {
  getMediaCredits,
  getMediaDetails,
  getMediaGenres,
  getMediaImages,
  getMediaRecommendations,
  getMediaVideos,
  getMediasList,
};
