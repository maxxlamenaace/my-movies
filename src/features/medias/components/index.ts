import CircularRate from './circular-rate/CircularRate';
import MediasAutoSwiper from './medias-auto-swiper/MediasAutoSwiper';
import MediasList from './medias-list/MediasList';
import MediaItem from './media-item/MediaItem';
import MediaBackgroundHeader from './media-background-header/MediaBackgroundHeader';
import MediaInfos from './media-infos/MediaInfos';
import CastSlide from './cast-slide/CastSlide';
import MediaVideosSlide from './media-videos-slide/MediaVideosSlide';
import MediaVideoItem from './media-video-item/MediaVideoItem';
import MediaImagesSlide from './media-images-slide/MediaImagesSlide';
import MediasGrid from './medias-grid/MediasGrid';
import FavoriteItem from './favorite-item/FavoriteItem';
import FavoritesGrid from './favorites-grid/FavoritesGrid';

export {
  CircularRate,
  MediasAutoSwiper,
  MediasList,
  MediaItem,
  MediaBackgroundHeader,
  MediaInfos,
  CastSlide,
  MediaVideosSlide,
  MediaVideoItem,
  MediaImagesSlide,
  MediasGrid,
  FavoriteItem,
  FavoritesGrid,
};
