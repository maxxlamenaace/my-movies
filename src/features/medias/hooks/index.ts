import { useMediaDetails } from './use-media-details/use-media-details';
import { useMediaGenres } from './use-media-genres/use-media-genres';
import { useAllMedias } from './use-all-medias/use-all-medias';
import { useMediasList } from './use-medias-list/use-medias-list';
import { useFavoriteMedias } from './use-favorite-medias/use-favorite-medias';

export { useMediaDetails, useAllMedias, useMediaGenres, useMediasList, useFavoriteMedias };
