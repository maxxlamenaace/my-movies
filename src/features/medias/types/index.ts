export type MediaType = 'movie' | 'tv';
export type MediaCategory = 'popular' | 'top_rated';
export type MediaImageType = 'poster' | 'backdrop';

export type Media = {
  id: number;
  poster_path?: string;
  adult: boolean;
  overview: string;
  genre_ids: number[];
  original_title: string;
  original_language: string;
  media_type?: MediaType;
  title?: string; // Only for movies
  name?: string; // Only for series
  backdrop_path?: string;
  popularity: number;
  vote_count: number;
  video: boolean;
  vote_average: number;
  release_date?: string; // Only for movies
  first_air_date?: string; // Only for series
};

export type MediaDetails = {
  id: number;
  budget: number;
  genres: Genre[];
  homepage: string;
  revenue: number;
  runtime: number;
  status: string;
  tagline: string;
} & Media;

export type MediaCredit = {
  id: number;
  adult: boolean;
  gender: number;
  known_for_department: string;
  name: string;
  original_name: string;
  popularity: number;
  profile_path: string;
  cast_id: number;
  character: string;
  credit_id: string;
  order: number;
};

export type MediaVideo = {
  id: number;
  type: string;
  official: boolean;
  name: string;
  site: string;
  size: number;
  published_at: Date;
  key: string;
};

export type MediaImage = {
  aspect_ratio: number;
  file_path: string;
  height: number;
  width: number;
};

/* export type ProductionCompany = {
  id: number;
  logo_path: string;
  name: string;
  origin_country: string;
};

export type ProductionCountry = {
  iso_3166_1: string;
  name: string;
};

export type SpokenLanguage = {
  iso_639_1: string;
  name: string;
}; */

export type Genre = {
  id: number;
  name: string;
};
