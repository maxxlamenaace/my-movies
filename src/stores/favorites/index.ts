export * from './types';
export * from './favorites';
