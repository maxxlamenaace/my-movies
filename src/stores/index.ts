import { useAppStore } from './app/app';
import { useThemeModeStore } from './theme-mode/theme-mode';
import { useFavoritesStore } from './favorites/favorites';

export { useAppStore, useFavoritesStore, useThemeModeStore };
