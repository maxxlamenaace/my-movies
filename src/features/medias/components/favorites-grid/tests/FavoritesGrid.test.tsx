import React from 'react';

import { BrowserRouter } from 'react-router-dom';
import { act, cleanup, render, renderHook } from '@testing-library/react';

import { mediaDetails } from '@/tests/factories';
import { FavoritesGrid } from '@/features/medias';
import { useFavoritesStore } from '@/stores';

describe('FavoritesGrid', () => {
  const media = mediaDetails();

  afterEach(() => {
    cleanup();
  });

  it('renders correctly', () => {
    const { result } = renderHook(() => useFavoritesStore());

    act(() => {
      result.current.setFavorites([{ mediaId: media.id, mediaType: 'movie' }]);
    });

    const { asFragment } = render(<FavoritesGrid medias={[media]} />, { wrapper: BrowserRouter });
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders with as many favorite grid items as needed', () => {
    const { result } = renderHook(() => useFavoritesStore());

    act(() => {
      result.current.setFavorites([{ mediaId: media.id, mediaType: 'movie' }]);
    });

    const { container } = render(<FavoritesGrid medias={[media]} />, { wrapper: BrowserRouter });

    expect(container.querySelectorAll('.favorite-grid-item')).toHaveLength(1);
  });
});
