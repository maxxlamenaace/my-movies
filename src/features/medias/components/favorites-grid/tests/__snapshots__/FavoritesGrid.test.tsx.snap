// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FavoritesGrid renders correctly 1`] = `
<DocumentFragment>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1 favorite-grid css-1rmx1e5-MuiGrid-root"
  >
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 MuiGrid-grid-sm-4 MuiGrid-grid-md-3 favorite-grid-item css-1vm8tgf-MuiGrid-root"
    >
      <div
        class="css-zgbp5c-MuiStack-root"
      >
        <a
          class="favorite-item"
          href="/medias/movie/1485624457"
        >
          <div
            class="MuiBox-root css-10py5dl"
          >
            <div
              class="media-back-drop MuiBox-root css-2e5d9x"
            />
            <button
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium media-play-btn css-3i9v1t-MuiButtonBase-root-MuiIconButton-root"
              tabindex="0"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-17slstj-MuiSvgIcon-root"
                data-testid="PlayArrowIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M8 5v14l11-7z"
                />
              </svg>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
            <div
              class="media-info MuiBox-root css-5bs0yh"
            >
              <div
                class="css-1okj3ks-MuiStack-root"
              >
                <div
                  class="MuiBox-root css-d2o2iw"
                >
                  <span
                    aria-valuenow="90"
                    class="MuiCircularProgress-root MuiCircularProgress-determinate MuiCircularProgress-colorSuccess css-zc2y38-MuiCircularProgress-root"
                    role="progressbar"
                    style="width: 50px; height: 50px; transform: rotate(-90deg);"
                  >
                    <svg
                      class="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
                      viewBox="22 22 44 44"
                    >
                      <circle
                        class="MuiCircularProgress-circle MuiCircularProgress-circleDeterminate css-oxts8u-MuiCircularProgress-circle"
                        cx="44"
                        cy="44"
                        fill="none"
                        r="20.2"
                        stroke-width="3.6"
                        style="stroke-dasharray: 126.920; stroke-dashoffset: 12.692px;"
                      />
                    </svg>
                  </span>
                  <div
                    class="MuiBox-root css-15i2x3e"
                  >
                    <div
                      class="MuiTypography-root MuiTypography-caption css-1dtdxxd-MuiTypography-root"
                    >
                      9
                    </div>
                  </div>
                </div>
                <p
                  class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
                />
                <p
                  class="MuiTypography-root MuiTypography-body1 css-qcvrn6-MuiTypography-root"
                >
                  title
                </p>
              </div>
            </div>
          </div>
        </a>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-fullWidth MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-fullWidth css-1fu7jd5-MuiButtonBase-root-MuiButton-root"
          id="favorite-item-delete-button"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-startIcon MuiButton-iconSizeMedium css-1d6wzja-MuiButton-startIcon"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="DeleteIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
              />
            </svg>
          </span>
          <span
            class="MuiTypography-root MuiTypography-button css-x0xkop-MuiTypography-root"
          >
            Delete
          </span>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
